### PR TITLE
DR 1440 add retry loop to dependency dao

### PR DESCRIPTION
**Pull request by RoriCremer**

### CODE REVIEW

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDependencyDao.java [hunk 1]
```diff
@@ -128,49 +130,57 @@ private void batchStoreSnapshotFileDependencies(
...
-                    }
-
-                    case 1: {
-                        // existing dependency; increment the reference count
-                        QueryDocumentSnapshot docSnap = documents.get(0);
-                        FireStoreDependency fireStoreDependency = docSnap.toObject(FireStoreDependency.class);
-                        fireStoreDependency.refCount(fireStoreDependency.getRefCount() + 1);
-                        setFutures.add(docSnap.getReference().set(fireStoreDependency));
-                        break;
+        for (int i = 0; i < RETRIES; i++) {
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDependencyDao.java]: Maybe I'm missing something, but I don't see how this loop exits early if no errors occur. Or is it OK to always run the body three times, even if no errors occur?

In the case where an error does occur, and the loop retries the body, is there no work needed to remove any futures that executed successfully and don't need to be retried?

**Review by fboulnois — COMMENTED** [review]
You mentioned you put this back to draft pending further investigation, correct?

**Comment by RoriCremer** [comment]
yes--I wanted to see if I could track down any other firestore requests that need a retry
I think the main place I realize they need to be added is in the delete workflow--and that should be something I can fold into this pr

**Review by ddietterich — APPROVED** [review]
lgtm

**Review by snf2ye — APPROVED** [review]

**Pull request merged**